main/src/ca/uwaterloo/flix/util/Validation.scala

```diff
@@ -45,19 +47,27 @@ sealed trait Validation[+T, +E] {
     *
     * Preserves the errors.
     */
-    // Deprecated. Use flatMapN instead.
+  // Deprecated. Use flatMapN instead.
   final def flatMap[U, A >: E](f: T => Validation[U, A]): Validation[U, A] = this match {
     case Validation.Success(input) => f(input) match {
       case Validation.Success(value) => Validation.Success(value)
+      case Validation.SuccessWithFailures(value, thatErrors) => Validation.SuccessWithFailures(value, errors #::: thatErrors)
       case Validation.Failure(thatErrors) => Validation.Failure(errors #::: thatErrors)
     }
+
+    case Validation.SuccessWithFailures(input, errors) => f(input) match {
+      case Validation.Success(value) => Validation.Success(value)
```

Comment:
This is wrong. We have to keep the `errors`.

let us see if you tests find it :)

Reply:
Are these not deprecated in favor of `flatMapN` and `mapN`?

Reply:
Yes, but SuccessWithFailure should never become Success, right?

You have done that correct everywhere else, as far as I can see :)